src/ngx_rewrite_driver_factory.cc

```diff
@@ -151,6 +151,10 @@ RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptions() {
   return options;
 }
 
+RewriteOptions* NgxRewriteDriverFactory::NewRewriteOptionsForQuery() {
+  return new NgxRewriteOptions(thread_system());
```

Comment:
Also, in NgxRewriteDriverFactory::NewRewriteOptions you are calling
  options->SetRewriteLevel(RewriteOptions::kCoreFilters);
whereas in mod_instaweb.cc's create_dir_config we are calling
  config->SetDefaultRewriteLevel(RewriteOptions::kCoreFilters);
which is 'weaker'.  That may also be contributing to this problem.  Does it work for you at least to change to setting the default rewrite-level instead of setting the rewrite level?